### PR TITLE
Added comment removal API

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Part of #278 

#### What's this PR do?
Updates the comments api to allow removal of comments by moderators.

Also reworked the `CommentFactory` because it hadn't been really used in a test yet and the previous iteration was impractical.

#### How should this be manually tested?
Go to `http://localhost:8063/api/v0/comments/:comment_id/` and `PATCH` with `{"removed": true}` and `{"removed": false}` and verify you see the expected state changes.

### CODE REVIEW

**Comment by aliceriot** [comment]
OK I can remove a comment, but it looks like it still comes back in the list of comments for a post. Is this the expected behavior?

**Comment by rhysyngsun** [comment]
@aliceriot if you're moderator, yes, because you need something to be able to re-approve.

**Comment by aliceriot** [comment]
aha that makes sense!

File: channels/serializers.py [hunk 1]
```diff
@@ -401,10 +410,18 @@ def create(self, validated_data):
     def update(self, instance, validated_data):
         if validated_data.get('comment_id'):
             raise ValidationError("comment_id must be provided via URL")
+
+        # import pdb ; pdb.set_trace()
```
Comment (aliceriot) [inline channels/serializers.py]: :thinking:

File: channels/utils.py [hunk 2]
```diff
@@ -87,6 +87,7 @@ def translate_praw_exceptions():
         # but I haven't seen any other causes for redirects
         raise NotFound() from exc
     except APIException as exc:
+        print(exc)
```
Comment (aliceriot) [inline channels/utils.py]: stray print?

**Comment by aliceriot** [comment]
ok cool, I also confirmed that, if I remove a comment as a moderator, as a non-moderator I see this:

removed_thing

**Comment by pdpinch** [comment]
Can we make it look like the design?

<image src="https://user-images.githubusercontent.com/20047260/31553774-883edf5c-b009-11e7-9926-1721ae7a391b.png">

**Comment by aliceriot** [comment]
As I understand it this is just covering the API work

**Comment by rhysyngsun** [comment]
@aliceriot feedback addressed

**Pull request merged**